src/Decoder.cpp

```diff
@@ -324,7 +289,7 @@ namespace nbs {
         uint64_t type = 0;
 
         try {
-            type = this->HashFromJsValue(typeSubtype.Get("type"), env);
+            type = Hash::FromJsValue(typeSubtype.Get("type"), env);
```

Comment:
Nice refactor moving this to a dedicated class. Same with the timestamp conversions.